backend/amd64/emit.mlp

```diff
@@ -1257,52 +1268,59 @@ let emit_instr fallthrough i =
           end
         end
       end
-  | Lop(Iextcall { func; alloc; }) ->
+  | Lop(Iextcall { func; alloc; stack_ofs }) ->
       add_used_symbol func;
-      if alloc then begin
+      if stack_ofs > 0 then begin
+        I.mov rsp r13;
+        I.lea (mem64 QWORD stack_ofs RSP) r12;
+        load_symbol_addr (Cmm.global_symbol func) rax;
+        emit_call (Cmm.global_symbol "caml_c_call_stack_args");
+        record_frame i.live (Dbg_other i.dbg);
+      end else if alloc then begin
         load_symbol_addr (Cmm.global_symbol func) rax;
         emit_call (Cmm.global_symbol "caml_c_call");
         record_frame i.live (Dbg_other i.dbg);
-        if system <> S_win64 then begin
-          (* TODO: investigate why such a diff.
-             This comes from:
-            http://caml.inria.fr/cgi-bin/viewvc.cgi?view=revision&revision=12664
-
-             If we do the same for Win64, we probably need to change
-             amd64nt.asm accordingly.
-          *)
-          I.mov (domain_field Domainstate.Domain_young_ptr) r15
-        end
       end else begin
-        emit_call (Cmm.global_symbol func)
+        I.mov rsp rbx;
+        cfi_remember_state ();
+        cfi_def_cfa_register "rbx";
+        (* NB: gdb has asserts on contiguous stacks that mean it
+           will not unwind through this unless we were to tag this
+           calling frame with cfi_signal_frame in it's definition. *)
+        (* CR xclerc for xclerc: according to the comment above, not
+           having the move will only hinder debugging. *)
+        (* I.mov (domain_field Domainstate.Domain_c_stack) rsp; *)
```

Comment:
If this is commented out, do we still need to spill `rsp` to `rbx` and if so why? This need to be coordinated with `destroyed_at_c_call` in `amd64/proc.ml` (as in ocaml-jst#243). This is the fast path of external calls, so important to optimize if  we can.